dm/syncer/syncer.go

```diff
@@ -1875,6 +1875,26 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 						// revert currentLocation to startLocation
 						currentLocation = startLocation
 					} else if op == pb.ErrorOp_Skip {
+						ec := eventContext{
```

Comment:
how about move  https://github.com/pingcap/tiflow/blob/fc70dbde873be0c09fa0f8f603772d2d95d59a8c/dm/syncer/syncer.go#L1923  to  https://github.com/pingcap/tiflow/blob/fc70dbde873be0c09fa0f8f603772d2d95d59a8c/dm/syncer/syncer.go#L1841

so that we can resue struce `ec`, in replaceErr logic, we can just update the filed in `ec`

Reply:
Many fields vary when we create this `ec`. I'm afraid that others may lose some fields' changes when they try to deal with this variable. I have removed all the unused fields.